### ISSUE TITLE
output.html.cname and routing from subdomain to github pages

### ISSUE THREAD

**Issue by VietThan** [rust-lang/mdBook#2166]
### Question

Hi,

### Problem
I am having problem trying to set up an mdbook to be served from GitHub Pages using GitHub Actions, and accessible though my own subdomain

The domain I want to be using for my book is: https://guides.vietnammun.org/

The mdbook to be served with Github Actions + GitHub Pages: https://github.com/VietnamMUN/book

Currently I'm encountering "Safari can't open the page because too many redirects occurred....This might occur if you open a page that is redirected to open another page which then is redirected to open the original page."

### What have I tried

I've followed the steps as detailed at [Managing a custom domain for your GitHub Pages site](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site).

I've also reviewed this [StackOverflow question](https://stackoverflow.com/questions/54059217/how-to-fix-domain-does-not-resolve-to-the-github-pages-server-error-in-github) but I don't have the same kind of error as the question.

I've added the following to `book.toml` as instructed by [mdBook](https://rust-lang.github.io/mdBook/format/configuration/renderers.html)'s documentation:
```toml
[output.html]
cname = "guides.vietnammun.org"
```

My GitHub repo setting has **Source** for **GitHub Actions** and **Custom domain** to `guides.vietnammun.org`. I know it says "DNS Check in Progress" but it used to be green and all good
<img width="1013" alt="Screenshot 2023-08-14 at 9 39 36 PM" src="https://github.com/rust-lang/mdBook/assets/42504605/a78187e7-50e8-40d5-a916-1b9ac6bc2ea6">

I registered my domain with CloudFlare and has the following DNS records
<img width="1058" alt="Screenshot 2023-08-14 at 9 57 39 PM" src="https://github.com/rust-lang/mdBook/assets/42504605/523e6654-b97e-4522-bdbf-93e306b32742">

I thought at first I just need to wait 24hrs, but that's a long time to wait and what if I'm wrong.

### Version

Locally
```shell
$ mdbook --version
mdbook v0.4.34
```

This is my GitHub Actions, which I took pretty much from the book and is using the latest version of mdbook.

```yaml
# Sample workflow for building and deploying a mdBook site to GitHub Pages
#
# To get started with mdBook see: https://rust-lang.github.io/mdBook/index.html
#
name: Deploy mdBook site to Pages

on:
  # Runs on pushes targeting the default branch
  push:
    branches: ["main"]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
permissions:
  contents: read
  pages: write
  id-token: write

# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
concurrency:
  group: "pages"
  cancel-in-progress: false

jobs:
  # Build job
  build:
    runs-on: ubuntu-latest
    env:
      MDBOOK_VERSION: 0.4.21
    steps:
      - uses: actions/checkout@v3
      # - name: Install mdBook
      #   run: |
      #     curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf -y | sh
      #     rustup update
      #     cargo install --version ${MDBOOK_VERSION} mdbook
      - name: Install latest mdbook
        run: |
          tag=$(curl 'https://api.github.com/repos/rust-lang/mdbook/releases/latest' | jq -r '.tag_name')
          url="https://github.com/rust-lang/mdbook/releases/download/${tag}/mdbook-${tag}-x86_64-unknown-linux-gnu.tar.gz"
          mkdir mdbook
          curl -sSL $url | tar -xz --directory=./mdbook
          echo `pwd`/mdbook >> $GITHUB_PATH
      - name: Build with mdBook
        run: mdbook build
      - name: Setup Pages
        id: pages
        uses: actions/configure-pages@v3
      - name: Upload artifact
        uses: actions/upload-pages-artifact@v2
        with:
          path: ./book

  # Deployment job
  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    needs: build
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v2
```

**Comment by VietThan** [comment]
I've removed and saved and got the green **DNS check successful**
<img width="759" alt="Screenshot 2023-08-14 at 11 11 52 PM" src="https://github.com/rust-lang/mdBook/assets/42504605/60f49610-029e-451b-a19f-20f4fbfd4fcf">

**Comment by VietThan** [comment]
Trying to reach https://guides.vietnammun.org on Safari, now it is: "Load cannot follow more than 20 redirections".

**Comment by VietThan** [comment]
Ok, so I couldn't get source for Github Pages and GitHub Actions to work. I've switched tracks and now my GitHub Action is pushing to a branch `gh-pages`, and GitHub Pages is set to source from the branch instead. 

I used the excellent example at https://github.com/peaceiris/actions-mdbook , thank you peaceiris.

All in all, probably could have everything setup in less than 24hours if I did this from the beginning. Hopefully this issue makes it easier for anyone else.
